### ISSUE TITLE
[BUG] EMR 6.2 with Rapids Accelerator not enabling DPP

### ISSUE THREAD

**Issue by krajendrannv** [NVIDIA/spark-rapids#1378]
**Describe the bug**
TPC-DS Q38 reads 3x files (1823 vs 366) when using Rapids Acceleratos

**Steps/Code to reproduce bug**
Use EMR 6.2 setup instructions for RAPIDS Accelerator
If we disable DPP for CPU like below
spark.conf.set('spark.sql.optimizer.dynamicPartitionPruning.enabled','false')
start = time.time(); spark.sql(q38).collect(); end = time.time(); print(end-start)
we see the same number of files for GPU and CPU being scanned (read)

**Expected behavior**
We need GPU reader to read the same number of files as CPU reader

**Environment details (please complete the following information)**
 - Environment location: EMR 6.2 on AWS with g4dn.2xlarge instance
 - Spark configuration settings related to the issue
"Classification":"spark-defaults",
      "Properties":{
     	"spark.plugins":"com.nvidia.spark.SQLPlugin", 
     	"spark.sql.sources.useV1SourceList":"",
     	"spark.executor.extraJavaOptions":"-Dai.rapids.cudf.prefer-pinned=true",
     	"spark.executor.resource.gpu.discoveryScript":"/usr/lib/spark/scripts/gpu/getGpusResources.sh",      	
     	"spark.submit.pyFiles":"/usr/lib/spark/jars/xgboost4j-spark_3.0-1.0.0-0.2.0.jar",
     	"spark.executor.extraLibraryPath":"/usr/local/cuda/targets/x86_64-linux/lib:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/compat/lib:/usr/local/cuda/lib:/usr/local/cuda/lib64:/usr/lib/hadoop/lib/native:/usr/lib/hadoop-lzo/lib/native:/docker/usr/lib/hadoop/lib/native:/docker/usr/lib/hadoop-lzo/lib/native",
     	"spark.rapids.sql.concurrentGpuTasks":"4",
     	"spark.executor.resource.gpu.amount":"1",
     	"spark.executor.cores":"8",
     	"spark.task.cpus ":"1",
     	"spark.task.resource.gpu.amount":"0.125",
     	"spark.rapids.memory.pinnedPool.size":"2G",
     	"spark.executor.memoryOverhead":"2G",
     	"spark.locality.wait":"0s",
     	"spark.sql.shuffle.partitions":"200",
     	"spark.sql.files.maxPartitionBytes":"256m",
      "spark.sql.adaptive.enabled":"false"

**Comment by tgravescs** [comment]
note that running this with Apache Spark 3.0.1 on yarn the DPP works just fine there, the number of files is way less with it configured to true then if you turn it off

**Comment by tgravescs** [comment]
reported to EMR team and they are looking at it.

**Comment by tgravescs** [comment]
After discussing with the EMR team, we decided to not do anything with this right now. This only occurs with AQE is off and on EMR AQE is on by default, we generally expect most people to be running with AQE on so they wouldn't hit this issue.

The underlying issue is a check was added with AQE disabled that runs after the Rapids columnar rule which ends up thinking it can't reuse the result from the broadcast exchange. To get around the issue we would require changes to the plugin and potentially Spark itself.  We should investigate it more if we think its important.
